pennylane/__init__.py

```diff
@@ -250,31 +261,77 @@ def grad(func, argnum=None):
         function with respect to the differentiable arguments, or, if specified,
         the arguments in ``argnum``.
     """
-    # pylint: disable=no-value-for-parameter
-    if argnum is not None:
-        # for backwards compatibility with existing code
-        # that manually specifies argnum
-        return _grad(func, argnum)
 
-    def _gradient_function(*args, **kwargs):
-        """Inspect the arguments for differentiability, and
-        compute the autograd gradient function with required argnums
-        dynamically.
+    def __init__(self, fun, argnum=None, *args, **kwargs):
```

Comment:
I'm curious why we now introduce `*args` and `**kwargs`?

Reply:
You're right, this can be removed 🙂 I think I was trying to be extra cautious when coding this up